openquake/hazardlib/gsim/abrahamson_2014.py

```diff
@@ -0,0 +1,520 @@
+        """
+        if rup.ztor >= 20.0:
+            return C['a15']
+        else:
+            return C['a15'] * rup.ztor / 20.0
+
+    def _get_z1pt0ref(self, vs30):
+        """
+        This computes the reference depth to the 1.0 km/s interface using
+        equation 18 at page 1042 of Abrahamson et al. (2014)
+        """
+        return 1/1000 * np.exp(-7.67/4.*np.log((vs30**4+610.**4) /
+                                               (1360.**4+610.**4)))
+
+    def _get_soil_depth_term(self, C, z1pt0, vs30):
+        """
+        Compute and return soil depth term.
+        See page 1042.
+        """
+        # Get reference z1pt0
+        z1ref = self._get_z1pt0ref(vs30)
+
+        # This is used for the calculation of the motion on reference rock
+        idx = z1pt0 < 0
+        z1pt0[idx] = z1ref[idx]
+        factor = np.log((z1pt0+0.01)/(z1ref+0.01))
+        # Here we use a linear interpolation as suggested in the 'Application
+        # guidelines' at page 1044
+        f2 = interpolate.interp1d(
+            [0.0, 150, 250, 400, 700, 1000, 2000],
+            [C['a43'], C['a43'], C['a44'], C['a45'], C['a46'], C['a46'],
+             C['a46']],
+            kind='linear')
+        return f2(vs30) * factor
+
+    def _get_regional_term(self, C, vs30, imt, rrup):
+        """
+        In accordance with Abrahamson et al. (2014) we assume as the default
+        region California. For the default case we don't apply a correction.
+        """
+        return 0.
+
+    def _get_stddevs(self, C, rup, sites, stddev_types, sa1180, dists):
+        """
+        Return standard deviations as described in paragraph 'Equations for
+        standard deviation', page 1046.
+        """
+        std_intra = self._get_intra_event_std(C, rup.mag, sa1180, sites.vs30,
+                                              sites.vs30measured, dists.rrup)
+        std_inter = self._get_inter_event_std(C, rup.mag, sa1180, sites.vs30)
+
+        stddevs = []
+        for stddev_type in stddev_types:
+            assert stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+            if stddev_type == const.StdDev.TOTAL:
+                stddevs.append(np.sqrt(std_intra ** 2 +
+                                       std_inter ** 2))
+            elif stddev_type == const.StdDev.INTRA_EVENT:
+                stddevs.append(std_intra)
+            elif stddev_type == const.StdDev.INTER_EVENT:
+                stddevs.append(std_inter)
+        return stddevs
+
+    def _get_intra_event_std(self, C, mag, sa1180, vs30, vs30measured, rrup):
+        """
+        Returns Phi as described at pages 1046 and 1047
+        """
+        phi_al = self._get_phi_al_regional(C, mag, vs30measured, rrup)
+        derAmp = self._get_derivative(C, sa1180, vs30)
+        phi_amp = 0.4
+        if any(phi_al**2 < phi_amp**2):
+            print 'phi_al:' % (phi_al)
+            print 'magnitude: %.2f' % (mag)
+            warnings.warn('sqrt argument < 0')
```

Comment:
Should this be a warning or an error? If this becomes a complex them then the resulting hazard curve will have complex values and the errors may propagate.